include/aws/common/mutex.h

```diff
@@ -36,10 +35,12 @@ struct aws_mutex {
 extern "C" {
 #endif
 
+AWS_COMMON_API struct aws_mutex aws_mutex_static_init (void);
```

Comment:
This is not ideal. That still requires code to be run to initialize the mutex, and it assumes that mutexes are moveable structures (which is probably the case for static initialized things, but might not always be true).

Instead do something like

```
#ifdef _WIN32
#define AWS_MUTEX_INIT SRWLOCK_INIT
#else
#define AWS_MUTEX_INIT PTHREAD_MUTEX_INITIALIZER
#endif
```

This makes it a compile time constant, so mutexes can be defined at the file level pre-initialized.

Reply:
This breaks the abstraction/ isolation of interface from implementation. I think that’s less ideal.

Reply:
Actually scratch that, this abstraction is already broken for mutex. Will update